### ISSUE TITLE
go-header-bar "title" is not changing when input variable "heading" is updated.

### ISSUE THREAD

**Issue by sudheepdivakargithub** [mobi/goponents#570]
**Describe the bug**
After the page is rendered, if we change the value passed to the @Input variable "heading", the same is not updated in the title of go-header-bar component.  

**To Reproduce**
1. Get a template rendered in the  go-header-bar component.
2. Update the heading input variable. 
3. Observe that the go-header component still shows the old value. The new value is not updated.

**Expected behavior**
The new value of the heading should get rendered in the component.

**Screenshots**

**Comment by grahamhency** [comment]
Hey @sudheepdivakargithub, thanks for submitting an issue. Can you provide a code example for how you're trying to change the `heading`. 

It seems like from this issue and the other issue you created (#571) that you're trying to change the content on the page without using Angular routing, which may be an anti-pattern of Angular.

Regardless, the reason this is happening is because the `goHeaderBar` directive only sets the `activeItem` on the `GoHeaderBarService` when it initializes. Looks like maybe we'll need to look into modifying it to update `onChanges`:
```
ngOnInit(): void {
  this.goHeaderBarService.activeItem.next({
    templateRef: this.template,
    title: this.heading,
    showBackArrow: this.showBackArrow,
    backButtonFn: this.handleBackButton
  });
}
```

**Comment by sudheepdivakargithub** [comment]
Hi Graham,

Yes, if the activeItem gets set onChanges hook, this should work.

Thanks,
--Sudheep--

**Comment by grahamhency** [comment]
Alternatively, you can set the re-render this template manually if you need to change it once rendered while on the same route.
In your `component.html`:
```
<ng-template
  #headerBar
  goHeaderBar
  [heading]="title"
  (handleBackButton)="goBackClicked()"
  [showBackArrow]="true">
</ng-template>
```
In your `component.ts`:
```
title: string = 'Dashboard';

@ViewChild('headerBar') headerBar: TemplateRef<any>;

constructor(private goHeaderBarService: GoHeaderBarService) { }

changeHeader(): void {
  this.goHeaderBarService.activeItem.next({
    templateRef: this.headerBar,
    title: this.title,
    showBackArrow: false
  });
}
```

**Comment by sudheepdivakargithub** [comment]
I did try this. But prod build was throwing "Module not found error" for GoHeaderBarService.

**Comment by grahamhency** [comment]
@sudheepdivakargithub you'll need to import it into your module.

**Comment by sudheepdivakargithub** [comment]
GoHeaderBarModule is already imported.
While I did a Typescript import import { GoHeaderBarService } from
'@tangoe/goponents/lib/components/go-header-bar/go-header-bar.service', prod
build gave me the error.

Thanks,
--Sudheep--

**Comment by grahamhency** [comment]
@sudheepdivakargithub ah, my bad. It looks like we're not exporting `GoHeaderBarService` in the `public_api` file which does not make it accessible to be used in your app. I will put up a PR to expose this. This should allow you to use this work around. We're actually going to be deploying another patch version tomorrow so we should be able to get this added in.

**Comment by grahamhency** [comment]
@sudheepdivakargithub, I've got a fix for this issue up for review. I am still looking into a solution for #571.. that one is tricky.

**Comment by grahamhency** [comment]
Closed with #581
